torchbox/torchbox.com
# Photo collage block - BE

[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1393406114)

### Description of Changes Made

This PR builds up on the homepage backend work (#117) to add a new photo collage block to the homepage and servicepage body StreamFields.

In keeping things DRY, the `ImageBlock` is refactored by defining a `BaseImageBlock` from which it inherits. The `PhotoCollageBlock` makes use of this `BaseImageBlock`, which mat also be handy elsewhere, if there's a need to specify custom alt text for an image.

**Additional notes**

- The FE is handled in #118. As much as possible, I have maintained the same field names used there. I have copied `tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/photo_collage_block.html` from there and modified it to account for alt text.
- I have added a commit (b24e7bd) to harmonize usage of blocks imported from `wagtail.blocks` in the `tbx.core.blocks` module, as there was a mix of, say, `blocks.CharBlock(...)` and `CharBlock(...)`, which was confusing. I suppose there could be other areas in the codebase where this phenomenon exists, but in order to keep the scope of this PR focused, I didn't check. This is something that can be done in a separate PR, perhaps as part of cleanup/pre-lauch activities. 

### How to Test

- Edit the homepage `body` in Wagtail Admin by adding a photo collage.
- test against the ACs
- you can verify the correct alt text is applied by using the browser dev tools to inspect the images

### Screenshots

<details>
  <summary>Wagtail Admin - HomePage</summary>

![image](https://github.com/torchbox/torchbox.com/assets/7713776/ecca4355-3f09-4bf8-a1f0-ba0a9a47e9f4)

</details>

<details>
  <summary>Template - HomePage</summary>

![image](https://github.com/torchbox/torchbox.com/assets/7713776/ea10a084-9f13-4c71-8554-9aba557293a4)

</details>

<details>
  <summary>Wagtail Admin - ServicePage</summary>

![image](https://github.com/torchbox/torchbox.com/assets/7713776/277ab3f3-6274-42af-91a5-70998457ac10)

</details>

### MR Checklist


#### Unit tests


#### Documentation


#### Browser testing

  - Latest version of Chrome on Ubuntu
  - Latest version of Firefox on Ubuntu

#### Data protection


#### Accessibility


#### Sustainability


#### Pattern library
